forth_src/v.fs

```diff
@@ -568,3 +570,8 @@ rom-kernal bufstart loadb
 ?dup 0= if reset-buffer else
 eof ! 0 eof @ c! then
 else drop then main-loop ;
+
+swap 
+to latest
+: v ;
+latest !

```

Comment:
I think it would be nice to have something like 

    MODULE
    ...
    : v ... ; PUBLIC
    ....
    ;MODULE

where `;MODULE` strips all words within the module that are not tagged as `PUBLIC`.

Also it would make sense to bring back good old `HIDE` , to remove words from the wordlist.

Reply:
Would `HIDE` set the hidden flag or edit the dictionary exactly like `DELETE`?  If the latter, MARKER is going to need a rework.

Reply:
There are other ways to do it, too...

    MARKER ---module---
    ...
    : V ... ; SHOW
    ...
    HIDE-TO ---marker---

Reply:
Humm... I didn't see `DELETE` until now, to be honest. I think actually what I mean is to rename `DELETE` to `HIDE`.

Reply:
I think it's not worth having `DELETE`/`HIDE` update markers. It is fine leaving it unsupported as long as it is mentioned in the manual.

Reply:
Working on this and trying to implement the `SHOW` element, and I'm finding myself in need of scratch space.  By definition, I shouldn't be using `PAD`.  Would it be safe to use $a000 for the entire span of `MODULE` to `;MODULE`?  I.e.: is it okay to add the restriction "the RAM block at $a000 is unavailable for use when compiling modules." ?

Reply:
You know, if my idea leads to such complications, I really think it's a terrible one!

Maybe just doing something real simple instead? E.g.

    MARKER ---EDITOR---

    DEFER V
 
    LATEST @
    ...
    :NONAME ... ; IS V
    ...
    LATEST !

Then no new implementation is needed, besides moving some words out of compat.fs.
In practice, I think this is really a practical and user-friendly solution, because the "public" words are placed in top of the file.

Reply:
It has to be acknowledged, this solution is not optimal size-wise because defer will waste some byte or two, but in this case I think the user-friendliness outweights that drawback.

Reply:
Instead of DEFER v, how about HEADER v?  Then no overhead.  Need a word to get the XT address of a word, but that's it.

Reply:
Actually, wasn't `DEFER` zero overhead anyway?

Reply:
I realize that `DEFER` modifying a `JSR` target has different behaviour than (what I'm calling) `DEFINES`.

If you get the execution token of a `DEFER`red word, you can call it multiple times and get different results if the deferred word changes.  If you get a `DEFINES`-ed execution token, it points to whatever code was assigned to the word at the time you got the XT, it will never change.

At least, I think that's the difference.

Reply:
`DEFINES` could be called `SHOW`, then,

```
: module: latest ;
: module; to latest ;
```

Which is dictionary bloat but syntactic sugar.

Reply:
Wouldn't this `DEFER` implementation work just fine?

    : DEFER HEADER IS ABORT ;

Reply:
I'm tired and I can't code or think much :)  But wouldn't it be enough that `DEFER` creates a wordlist entry using `HEADER` and then sets the xt to `ABORT`?

Reply:
...conversedly, all that `IS` and `DEFER!` need to do is to update xt of a wordlist entry?

Reply:
> If you get the execution token of a DEFERred word, you can call it multiple times and get different results if the deferred word changes. If you get a HIDE-n execution token, it points to whatever code was assigned to the word at the time you got the XT, it will never change.

I guess the question is: do you ever expect that a user may want to `'` a deferred word in expectation that they can keep that token, `IS` the word, then call that token and get the new value?

```
defer x
: foo [`] x ;
x is bar \ foo execute will run bar
header y
: baz [`] y ;
y show bar \ baz execute will run abort"
```

Reply:
The more I think about it, the more SHOW behaviour (modifying the XT of a dictionary entry) is distinct from DEFER/IS.  Consider:
```
defer foo
: bar foo ;
... some code later ...
baz is foo
```
This pattern is typical of `DEFER`, you declare a word so that you can reference it before the behaviour is defined.  If `foo`'s XT was `abort`, baz would compile to a call to abort.  The indirection is essential.

When exporting from a hidden space, the goal is not to reference the word before its behavioural definition, it is to alias some hidden code to a public word for later reference.  While the below code could use `DEFER` and only incur overhead, the above code could not use `SHOW`.
```
header foo
hide:
: bar ;
: baz bar ;
` baz show foo
;hide
```